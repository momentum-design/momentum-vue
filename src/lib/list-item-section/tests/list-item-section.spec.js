import { mount } from '@vue/test-utils';
import ListItemSection from '../index.vue';

describe('ListItemSection', () => {
  it('should match snapshot', () => {
    const wrapper = mount(ListItemSection);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render one ListItemSection', () => {
    const wrapper = mount(ListItemSection);

    expect(wrapper.findAll('.md-list-item__center').length).toEqual(1);
  });

  it('should handle position prop (left)', () => {
    const wrapper = mount(ListItemSection, {
      propsData: {
        position: 'left'
      }
    });

    expect(wrapper.findAll('.md-list-item__left').length).toEqual(1);
  });

  it('should handle position prop (right)', () => {
    const wrapper = mount(ListItemSection, {
      propsData: {
        position: 'right'
      }
    });

    expect(wrapper.findAll('.md-list-item__right').length).toEqual(1);
  });

  it('should handle class prop', () => {
    const wrapper = mount(ListItemSection, {
      propsData: {
        class: 'right'
      }
    });

    expect(wrapper.findAll('.right').length).toEqual(1);
  });

  it('should render children', () => {
    const wrapper = mount(ListItemSection, {
      slots: {
        default: '<div>Test</div>'
      }
    });

    expect(wrapper.element.__vue__.$children.length).toEqual(1);
  });
});

