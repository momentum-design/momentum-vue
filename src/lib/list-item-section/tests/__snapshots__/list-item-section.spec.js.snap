// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemSection should match snapshot 1`] = `<div class="md-list-item__center"></div>`;
